app/workers/fetch_keywords_worker.rb

```diff
@@ -4,4 +4,4 @@ class FetchKeywordsWorker
   def perform(report_id)
     Api::V1::Reports::FetchKeywords.call(report_id)
```

Comment:
Sidekiq has a batch feature that could be used: https://github.com/sidekiq/sidekiq/wiki/Batches#huge-batches